packages/dev/core/src/FlowGraph/Blocks/Execution/ControlFlow/flowGraphForLoopBlock.ts

```diff
@@ -86,6 +93,10 @@ export class FlowGraphForLoopBlock extends FlowGraphExecutionBlockWithOutSignal
             }
         }
 
+        if (this.config?.incrementIndexWhenLoopDone) {
+            this.index.setValue(new FlowGraphInteger(getNumericValue(this.index.getValue(context)) + 1), context);

```

Comment:
I think it should be `getNumericValue(this.index.getValue(context)) + step`?

Also (I can't comment on a line that has not been modified):
```typescript
    if (i > FlowGraphForLoopBlock.MaxLoopIterations) {
```
If `MaxLoopIterations` is really a maximum number of loop iterations, shouldn't it take into account `step`? Something like `if (i > FlowGraphForLoopBlock.MaxLoopIterations * step)`.

Reply:
Good catch on the step, i'll change it.

The second line - of course! will change that.

Reply:
done :-)